arch/posix/core/fatal.c

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (c) 2016 Intel Corporation
+ * Copyright (c) 2017 Oticon A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*Started as a copy of the nios2 one with some prunning*/
+
+#include <kernel.h>
+#include <arch/cpu.h>
+#include <kernel_structs.h>
+#include <misc/printk.h>
+#include <inttypes.h>
+#include "posix_soc_if.h"
+
+const NANO_ESF _default_esf = {
+	0xdeadbaad
+};
+
+/**
+ *
+ * @brief Kernel fatal error handler
+ *
+ * This routine is called when a fatal error condition is detected by either
+ * hardware or software.
+ *
+ * The caller is expected to always provide a usable ESF.  In the event that the
+ * fatal error does not have a hardware generated ESF, the caller should either
+ * create its own or call _Fault instead.
+ *
+ * @param reason the reason that the handler was called
+ * @param pEsf pointer to the exception stack frame
+ *
+ * @return This function does not return.
+ */
+FUNC_NORETURN void _NanoFatalErrorHandler(unsigned int reason,
+		const NANO_ESF *esf)
+{
+#ifdef CONFIG_PRINTK
+	switch (reason) {
+	case _NANO_ERR_CPU_EXCEPTION:
+	case _NANO_ERR_SPURIOUS_INT:
+		break;
+
+	case _NANO_ERR_INVALID_TASK_EXIT:
+		printk("***** Invalid Exit Software Error! *****\n");
+		break;
+
+
+	case _NANO_ERR_ALLOCATION_FAIL:
+		printk("**** Kernel Allocation Failure! ****\n");
+		break;
+
+	case _NANO_ERR_KERNEL_OOPS:
+		printk("***** Kernel OOPS! *****\n");
+		break;
+
+	case _NANO_ERR_KERNEL_PANIC:
+		printk("***** Kernel Panic! *****\n");
+		break;
+
+#ifdef CONFIG_STACK_SENTINEL
+	case _NANO_ERR_STACK_CHK_FAIL:
+		printk("***** Stack overflow *****\n");
+		break;
+#endif
+	default:
+		printk("**** Unknown Fatal Error %u! ****\n", reason);
+		break;
+	}
+
+#endif
+
+	void _SysFatalErrorHandler(unsigned int reason,
+			const NANO_ESF *pEsf);
+	_SysFatalErrorHandler(reason, esf);
+}
+
+#if defined(CONFIG_EXTRA_EXCEPTION_INFO) && defined(CONFIG_PRINTK) \
+		&& defined(ALT_CPU_HAS_EXTRA_EXCEPTION_INFO)
+static char *cause_str(u32_t cause_code)
```

Comment:
don't see why we need this since we are not invoking x86 CPU exceptions

Reply:
The whole fatal.c needs a cleanup (I started copying the nios one, but most of it was not necessary). => added to list of TODOs